ChildProject/pipelines/metrics.py

```diff
@@ -1,31 +1,71 @@
 from abc import ABC, abstractmethod
+import os
 import argparse
 import datetime
 import multiprocessing as mp
 import numpy as np
 import pandas as pd
 from typing import Union, List
+import yaml
+from git import Repo
+from git.exc import InvalidGitRepositoryError
 
 import ChildProject
 from ChildProject.pipelines.pipeline import Pipeline
 
-pipelines = {}
+import ChildProject.pipelines.metricsFunctions as metfunc #used by eval
+from ..utils import TimeInterval, time_intervals_intersect
 
+pipelines = {}
 
 class Metrics(ABC):
     def __init__(
         self,
         project: ChildProject.projects.ChildProject,
+        metrics_list: pd.DataFrame,
         by: str = "recording_filename",
         recordings: Union[str, List[str], pd.DataFrame] = None,
         from_time: str = None,
         to_time: str = None,
         rec_cols: str = None,
         child_cols: str = None,
+        period: str = None,
+        threads: int = 1,
     ):
 
         self.project = project
         self.am = ChildProject.annotations.AnnotationManager(self.project)
+        self.threads = int(threads)
+        
+        #check that the callable column is either a callable function or a string that can be evaluated as being part of the list of metrics in ChildProject/pipelines/metricsFunctions.py
+        def check_callable(row):
+            if callable(row["callable"]): return row["callable"]
+            if isinstance(row["callable"],str):
+                try:
+                    f = eval("metfunc."+row["callable"])
```

Comment:
Maybe safer do this instead of `eval':

```
f = getattr(metfunc, row['callable'])
```

Reply:
for sure, knew there was a better way of doing this!
 7daaba6